### DISCUSSION TITLE
Hysteria2 not working on iphone

### DISCUSSION THREAD

**Discussion by shopliftersoftheworldunite** [apernet/hysteria#1112] [General]
Hello,
I started using Hysteria2  but I ran into unique problem, I tried it on various servers and even two active servers at once for comparison, and they all yield the same result the servers seem to stop working on iPhone after a few days they are great on every other platform and no alteration has been made prior to the incident and yet hysteria2 suddenly stops working on iPhone if anyone had faced the same problem or have any idea why it occurs please do let me know.

**Comment by meower1** [comment]
have you tried using it on shadowrocket? could you specify what client you're using?

**Reply by shopliftersoftheworldunite** [reply]
morning, yes.
the problem seems to be two things, 1.luck, depending on the server you get while purchasing,
2.obfs seems to affect the outcome Iphones sometimes work better without it.
and I haven't faced problems recently

**Reply by meower1** [reply]
yes i've had similar experience, same configuration on two different servers seems to have different impacts as well obfs does degrade the experience. was hoping they would fix it in the newer versions but it appears it still exists. i'd suggest trying out sing-box based ios apps to see which one would work

**Comment by shakibamoshiri** [comment]
what client did you use for iPhone ?
Make sure  your certificate has not been expired 
_Hy2 has no issue with no device , works will on all of them_

**Reply by shopliftersoftheworldunite** [reply]
Shadowrocket,
the problem seems to be two things, 1.luck, depending on the server you get while purchasing,
2.obfs seems to affect the outcome Iphones sometimes work better without it.
and I haven't faced problems recently
